### PR TITLE
Introduce new max_sub_interval config for integral sensor

### PR DESCRIPTION
**Pull request by ronweikamp**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/110685
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by erkr — COMMENTED** [review]
Clear to me, thx

File: source/_integrations/integration.markdown [hunk 1]
```diff
@@ -22,10 +22,7 @@ This integrations provides the [Riemann sum](https://en.wikipedia.org/wiki/Riema
 of the values provided by a source sensor. The Riemann sum is an approximation
 of an **integral** by a finite sum.
 
-The integration sensors are updated upon changes of the **source**. Fast
-sampling source sensors provide more accurate results. In this implementation, the
```
Comment (dgomes) [inline source/_integrations/integration.markdown:26]: I would not remove this information as it still applies!

**Review by dgomes — DISMISSED** [review]

**Comment by AleXSR700** [comment]
Would it not be clearer to call it "Minimum update frequency" rather than "Maximum update duration" or "Max sub-interval"? It is not really a duration but rather a frequency or interval. But it would also be the minimum interval, i.e. update "at least every" and not "at most every" x minutes.

File: source/_integrations/integration.markdown (hunk 1 above)
Comment (ronweikamp) [reply source/_integrations/integration.markdown:26]: What is a "fast sampling source sensor"? And how does it provide "more accurate results"? And more accurate than what?

**Comment by ronweikamp** [comment]
> Would it not be clearer to call it "Minimum update frequency" rather than "Maximum update duration" or "Max sub-interval"? It is not really a duration but rather a frequency or interval. But it would also be the minimum interval, i.e. update "at least every" and not "at most every" x minutes.

It is a duration (an interval in time). Duration and frequency are inversely related here. A max update duration means a minimum update frequency. I've added this implication! Hopefully this makes it more clear?

File: source/_integrations/integration.markdown (hunk 1 above)
Comment (dgomes) [reply source/_integrations/integration.markdown:26]: fast sampling means you have more samples of the underlying continuous function (the real physical phenomena), therefore a rienmann sum will be able to better approach the continuous function (regardless of method) if you have more samples.

File: source/_integrations/integration.markdown [hunk 2]
```diff
@@ -91,17 +91,19 @@ method:
...
-In case you expect that your source sensor will provide several subsequent values that are equal, you should opt for the `left` method to get accurate readings.
-
 The unit of `source` together with `unit_prefix` and `unit_time` is used to generate a unit for the integral product (e.g. a source in `W` with prefix `k` and time `h` would result in `kWh`). Note that `unit_prefix` and `unit_time` are _also_ relevant to the Riemann sum calculation. 
 
 ## Integration method
 
-Riemann Sum is a approximation of an integral by a finite sum and is therefore intrinsically inaccurate, nonetheless, depending on the method used, values can be more or less accurate.
+Riemann Sum is a approximation of an integral by a finite sum and is therefore intrinsically inaccurate, nonetheless, values can be more or less accurate.
 
-Regardless of the method used the integration will be more accurate if the source updates more often. If your source is not updated, neither will the Riemann Sum sensor, as all this integration does is calculate the next step in the event of a source update.
```
Comment (dgomes) [inline source/_integrations/integration.markdown:104]: "Regardless of the method used the integration will be more accurate if the source updates more often. "

Still applies...

File: source/_integrations/integration.markdown [hunk 3]
```diff
@@ -91,17 +92,19 @@ method:
...
+  type: time
 {% endconfiguration %}
 
-In case you expect that your source sensor will provide several subsequent values that are equal, you should opt for the `left` method to get accurate readings.
-
 The unit of `source` together with `unit_prefix` and `unit_time` is used to generate a unit for the integral product (e.g. a source in `W` with prefix `k` and time `h` would result in `kWh`). Note that `unit_prefix` and `unit_time` are _also_ relevant to the Riemann sum calculation. 
 
 ## Integration method
 
-Riemann Sum is a approximation of an integral by a finite sum and is therefore intrinsically inaccurate, nonetheless, depending on the method used, values can be more or less accurate.
```
Comment (dgomes) [inline source/_integrations/integration.markdown:102]: by removing "depending on the method used, " this sentence became strange due to redundancy of "inaccurate"

**Comment by AleXSR700** [comment]
> It is a duration (an interval in time). Duration and frequency are inversely related here. A max update duration means a minimum update frequency. I've added this implication! Hopefully this makes it more clear?

It is clear enough to me but for me an update duration is how long an update is performed. A duration is an amount of time, so kind of a sum.
Here it is really an interval or frequency because nothing is done inbetween. Duration means that actions are being performed during that from start to end of duration. Interval means it is done at certain times.
Here, to me, it is an interval or frequency.

As I said, I care more about the merge but wanted to comment that is would be more clear as "frrequency" or "interval".

File: source/_integrations/integration.markdown [hunk 4]
```diff
@@ -22,10 +22,8 @@ This integrations provides the [Riemann sum](https://en.wikipedia.org/wiki/Riema
 of the values provided by a source sensor. The Riemann sum is an approximation
 of an **integral** by a finite sum.
 
-The integration sensors are updated upon changes of the **source**. Fast
-sampling source sensors provide more accurate results. In this implementation, the
-default is the Trapezoidal method, but Left and Right methods can optionally
-be used.
+The integration sensors are updated upon changes of the source and optionally based on
```
Comment (c0ffeeca7) [inline source/_integrations/integration.markdown]:
```suggestion
The integration sensors are updated whenever the source changes and, optionally, based on a predefined
```

File: source/_integrations/integration.markdown [hunk 5]
```diff
@@ -22,10 +22,8 @@ This integrations provides the [Riemann sum](https://en.wikipedia.org/wiki/Riema
...
+time. Fast sampling source sensors provide more accurate results.
```
Comment (c0ffeeca7) [inline source/_integrations/integration.markdown]:
```suggestion
time interval. Source sensors with higher sampling frequency provide more accurate results.
```
tiny tweak. feel free to ignore if I misunderstood this.

File: source/_integrations/integration.markdown [hunk 6]
```diff
@@ -41,6 +39,9 @@ Metric prefix:
   description: Metric unit to prefix the integration result.
 Integration time:
   description: SI unit of time to integrate over.
+Max sub-interval:
+  description: Applies time based integration if the source did not change for this duration. This implies that at least every 'max sub-interval' the integral is updated. Use 0 for no time based updates.
```
Comment (c0ffeeca7) [inline source/_integrations/integration.markdown]:
```suggestion
  description: Applies time-based integration if the source did not change for this duration. This implies that at least every `max sub-interval`, the integral is updated. If you don't want time-based updates, enter 0.
```

File: source/_integrations/integration.markdown [hunk 7]
```diff
@@ -91,17 +92,19 @@ method:
   required: false
   type: string
   default: trapezoidal
+max_sub_interval:
+  description: "Applies time based integration if the source did not change for this duration. This implies that at least every `max_sub_interval` the integral is updated. Use 0 for no time based updates."
```
Comment (c0ffeeca7) [inline source/_integrations/integration.markdown]:
```suggestion
  description: "Applies time-based integration if the source did not change for this duration. This implies that at least every `max sub-interval`, the integral is updated. If you don't want time-based updates, enter 0."
```

File: source/_integrations/integration.markdown [hunk 8]
```diff
@@ -91,17 +92,19 @@ method:
...
 {% endconfiguration %}
 
-In case you expect that your source sensor will provide several subsequent values that are equal, you should opt for the `left` method to get accurate readings.
-
 The unit of `source` together with `unit_prefix` and `unit_time` is used to generate a unit for the integral product (e.g. a source in `W` with prefix `k` and time `h` would result in `kWh`). Note that `unit_prefix` and `unit_time` are _also_ relevant to the Riemann sum calculation. 
 
 ## Integration method
 
-Riemann Sum is a approximation of an integral by a finite sum and is therefore intrinsically inaccurate, nonetheless, depending on the method used, values can be more or less accurate.
+Riemann Sum is a approximation of an integral by a finite sum and is therefore intrinsically inaccurate, nonetheless, values can be more or less accurate.
```
Comment (c0ffeeca7) [inline source/_integrations/integration.markdown]:
```suggestion
The Riemann Sum is an approximation of an integral by a finite sum and is therefore intrinsically inaccurate. Nonetheless, values can be more or less accurate.
```

File: source/_integrations/integration.markdown [hunk 9]
```diff
@@ -91,17 +92,19 @@ method:
...
 
-Regardless of the method used the integration will be more accurate if the source updates more often. If your source is not updated, neither will the Riemann Sum sensor, as all this integration does is calculate the next step in the event of a source update.
+The integration method defines how to calculate the area under the source sensor when it changes. Regardless of the method used the integration will be more accurate if the source updates more often. The config `max_sub_interval` can be used to trigger integration when the source sensor is constant.
```
Comment (c0ffeeca7) [inline source/_integrations/integration.markdown]:
```suggestion
The integration method defines how to calculate the area under the source sensor when it changes. Regardless of the method used, the integration will be more accurate if the source updates more often. The config `max_sub_interval` can be used to trigger integration when the source sensor is constant.
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]
Tweaks

File: source/_integrations/integration.markdown (hunk 8 above)
Comment (dgomes) [reply source/_integrations/integration.markdown]: `Nonetheless, values can be more or less accurate. `

As a user, I would ask why (this is basically all forum posts)

File: source/_integrations/integration.markdown [hunk 10]
```diff
@@ -91,17 +92,19 @@ method:
...
 {% endconfiguration %}
 
-In case you expect that your source sensor will provide several subsequent values that are equal, you should opt for the `left` method to get accurate readings.
-
 The unit of `source` together with `unit_prefix` and `unit_time` is used to generate a unit for the integral product (e.g. a source in `W` with prefix `k` and time `h` would result in `kWh`). Note that `unit_prefix` and `unit_time` are _also_ relevant to the Riemann sum calculation. 
 
 ## Integration method
 
-Riemann Sum is a approximation of an integral by a finite sum and is therefore intrinsically inaccurate, nonetheless, depending on the method used, values can be more or less accurate.
+The Riemann Sum is an approximation of an integral by a finite sum and is therefore intrinsically inaccurate. Nonetheless, values can be more or less accurate.
```
Comment (c0ffeeca7) [inline source/_integrations/integration.markdown]:
```suggestion
The Riemann Sum is an approximation of an integral by a finite sum and is therefore intrinsically inaccurate. Nonetheless, depending on the method used, values can be more or less accurate.
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @ronweikamp :+1:

**Pull request merged**